File: thain-core/src/main/java/com/xiaomi/thain/core/entity/ThainUser.java
/*
 * Copyright (c) 2019, Xiaomi, Inc.  All rights reserved.
 * This source code is licensed under the Apache License Version 2.0, which
 * can be found in the LICENSE file in the root directory of this source tree.
 */
package com.xiaomi.thain.core.entity;

import lombok.AllArgsConstructor;

import javax.annotation.Nullable;

/**
 * Date 2019/8/9 下午3:42
 *
 * @author liangyongrui@xiaomi.com
 */
@AllArgsConstructor
public class ThainUser {
    public final long id;
    @Nullable
    public final String userId;
    @Nullable
    public final String userName;
    @Nullable
    public final String passwordHash;
    @Nullable
    public final String email;
    public final boolean admin;
}

File: thain-core/src/main/kotlin/com/xiaomi/thain/core/entity/ThainUser.kt
package com.xiaomi.thain.core.entity

/**
 * Date 2019/8/9 下午3:42
 *
 * @author liangyongrui@xiaomi.com
 */
class ThainUser(
        val id: Long,
        val userId: String?,
        val userName: String?,
        val passwordHash: String?,
        val email: String?,
        val admin: Boolean
)
